quinn/src/endpoint.rs

```diff
@@ -523,7 +536,10 @@ impl State {
                                 .send(ConnectionEvent::Proto(event));
                         }
                     }
-                    Transmit(t) => self.outgoing.push_back(udp_transmit(t)),
+                    Transmit(t) => {
+                        self.outgoing.push_back(udp_transmit(t));
+                        self.transmit_queue_size.fetch_add(1, Ordering::Relaxed);
```

Comment:
The counter should be incremented when events are pushed onto the `events` queue by the connection, so that we count packets that are currently in the queue. That's why an atomic is needed.

Reply:
Alternatively, if we're only concerned about packets queued by the endpoint, trusting in congestion/flow control/pacing for connection-generated packets, we could just switch to a non-atomic counter that's fully internal to `Endpoint`.

Reply:
I think we do need to take consideration of connection generated packets for the connections in initial state -- I have found they could generate large amount of packets fast enough to generate a build up in the outgoing queue as well. So I think a Atomic variable is better for this purpose in the future and also for back porting to older version like 0.9.x.